src/base.ts

```diff
@@ -66,7 +107,17 @@ export default class KintoClientBase {
      * The remote server base URL.
      * @type {String}
      */
-    this.remote = remote;
+    let version;
+    try {
+      version = remote.match(/\/(v\d+)\/?$/)![1];
+    } catch (err) {
+      throw new Error("The remote URL must contain the version: " + remote);
+    }
+    if (version !== SUPPORTED_PROTOCOL_VERSION) {
+      throw new Error(`Unsupported protocol version: ${version}`);
+    }
+    this._remote = remote;
+    this._version = version;
```

Comment:
Why did you have to inline this code instead of using the setter?

Reply:
I have a feature called "strict initialization checks" enabled that requires class properties to be assigned in the constructor. However, we were actually always assigning anyway, it was just that the compiler couldn't determine that. I learned about a feature called the definite assignment assertion, so I removed the inline function and used those instead to tell the compiler that we are, in all circumstances, assigning those properties.